### ISSUE TITLE
Updating install documentation

### ISSUE THREAD

**Issue by nagyrobir** [talend-spatial/talend-spatial#87]
Hei All!

I would request an update on the documentation available for installation of Talend Spatial Library. With versioning of the three most important elements that are required to run Talend Spatial Library. I am facing difficulties in setting up this tool, by getting many error messages that refer to ogr, and i kindly ask you to help me out.

1. Currently on [(https://sourceforge.net/projects/sdispatialetl/files/screencast/)] the latest version is 7.1.1. I would assume then that one would need TOS that matches this version?
2. Which TOS does one need to download to make it work? Talend Big Data / Talend Data Integration / etc.?
3. It is unclear which version of GDAL/OGR is compatible with Talend Spatial Library 7.1.1?
4. Clearer description of what parameters should be set. I have read some posts form the internet but it is a bit unclear from the readme doc as well.

Is the library available here for TOS 7.3 here? can it be downloaded from here and used?

If you are ok with creating a good description, i would like to make some youtube videos of how to set it up Talend Spatial Library and how to make it work. I think there is a huge community out there who is eagerly is looking for a substitute for FME. I understand the amazing job you guys have done, and i would like to spread the word to create a greater community and raising awareness for this nice tool.

Robert Nagy

**Comment by fxprunayre** [comment]
> * Currently on [(https://sourceforge.net/projects/sdispatialetl/files/screencast/)] the latest version is 7.1.1. I would assume then that one would need TOS that matches this version?

No. It has to be a version at least equal but can be a newer version of Talend Studio (but it may not work if major changes are introduced in Studio).



> * Which TOS does one need to download to make it work? Talend Big Data / Talend Data Integration / etc.?

On my side, I'm usually on TDI or ESB but there is no specific dependencies on a particular Studio

> It is unclear which version of GDAL/OGR is compatible with Talend Spatial Library 7.1.1?

In theory, it can be customized. You can update `gdal.jar` in the Studio to match the version you need. A JAR file is provided by default https://github.com/talend-spatial/talend-spatial/tree/7.3/main/plugins/org.talend.sdi.designer.components/components/sGdalInfoInput. And the JAR in Studio have to match the version of GDAL in the path.

As far as there is no major changes in GDAL it should work fine. An example of issue that can happen when a GDAL internal change https://github.com/talend-spatial/talend-spatial/commit/e73d1310436c98142c200e1c690029c62d449937

> Clearer description of what parameters should be set. I have read some posts form the internet but it is a bit unclear from the readme doc as well.

For GDAL, the doc on https://github.com/talend-spatial/workspace-metadata-crawler is quite complete with different cases.


> Is the library available here for TOS 7.3 here? can it be downloaded from here and used?

The spatial module can be loaded with 7.3.
See https://github.com/talend-spatial/talend-spatial/issues/80#issuecomment-623467558


Any help on better doc & description are welcomed.

**Comment by nagyrobir** [comment]
I cannot make this work. I work on a Windows 7 machine.

Downloads:
- I have downloaded Talend Spatial Library from https://sourceforge.net/projects/sdispatialetl/files/screencast/
- Have downloaded GDAL from http://download.gisinternals.com/sdk/downloads/release-1900-x64-gdal-3-1-2-mapserver-7-6-1/gdal-301-1900-x64-core.msi
- have downloaded talend DI from https://sourceforge.net/projects/talend-studio/files/Talend%20Open%20Studio/7.3.1/


- I have created the 3 paths exactly as described in the documentation for GDAL
- I have installed all required packages for TOS DI
- I have copied and pasted the contents of the Talend Spatial Library  into TOS/Plugins library
- I have edited under the configuration folder the config file and added the following under osgi.bundles
      o    ,org.talend.libraries.sdi-7.1.1@4,org.talend.sdi.designer.components-7.1.1@4,org.talend.sdi.designer.routines- 
             7.1.1@4,org.talend.sdi.repository.ui.actions.metadata-7.1.1@4,org.talend.sdi.repository.ui.actions.metadata.ogr- 
             7.1.1@4,org.talend.sdi.workspace.spatial-7.1.1@4


 - i have downloaded the workspace-demo (the one that you show on the tutorial from a few years ago) that throws  an error similar with #86 

```
org.talend.designer.runprocess.ProcessorException: Routines "GeometryOperation" has compile errors
Error Line: 1
Detail Message: The type org.opengis.referencing.crs.CoordinateReferenceSystem cannot be resolved. It is indirectly referenced from required .class files
```

- I can see all the tools that are available in the tools palette but i cannot used them. I tried to read in a PostGis Table with sPostgisInput connected it to a tLogRow and got the same error as described above.

- I have tried to copy the gdal.jar from C:\Program Files\GDAL\java to plugins\org.talend.sdi.repository.ui.actions.metadata.ogr-7.1.1\lib and plugins\org.talend.sdi.designer.components-7.1.1\components\sGdalInfoInput

- I have tested it on multiple versions of TOS DI and it only partially works on talend 7.1.1. where i get stuck on another error. 

What am i doing wrong? i tried my best.

**Comment by jlhgis** [comment]
We are having similar issues with GDAL and getting this to run properly.  Is there a listserv where we can get some better feedback on how to get configured properly? Not seeing much of anything in the main Talend community.  We have been really trying to get this set up on a Windows server and are not able to resolve the GDAL issues.  Would hate to abandon this tool as it could be so very promising, but the lack of current docs and detailed GDAL support info is not helping the decision.

**Comment by archaeogeek** [comment]
I'm not sure if this is any help, but I have had good success setting this up in docker, using a headless ubuntu server and then connecting to it with a vnc. I had the same problems with versions, but having upgraded Talend and Talend Spatial to 7.4.1 everything works just fine.

**Comment by fxprunayre** [comment]
@archaeogeek maybe the docker setup you have is something that you can share ? as quite some users have difficulties to set it up. That could help reproducing a working setup easily. Let us know. Thanks

**Comment by archaeogeek** [comment]
Sure- it's not perfect but it works for me. I could commit the dockerfile to a docker folder in the repository, with some usage instructions? Would that work?
